### ISSUE TITLE
Buffer -> Window assignments lost after closing frame when using Emacs daemon

### ISSUE THREAD

**Issue by mmeier86** [syl20bnr/spacemacs#13811]
#### Description :octocat:
I'm using emacs in daemon mode, also storing my layout and automatically
loading it on emacs startup.

Now, lets say I create a new layout and open `.emacs.d/README.md` in it.
Then, I quit the frame with `<SPC> q f`. The emacs server stays running.

Now, I start a new frame with `emacsclient -c`. My newly created layout
is still there, as expected. But if I now switch to that new layout,
I don't have the `.emacs.d/README.md` buffer open, but instead I see
the `scratch` buffer opened in the layout's only window. Pressing
`<SPC> b b` actually shows me that the buffer is still open, but it's
not loaded into the window.

Conversely, if I kill the server too, by pressing `<SPC> q q` instead
of just killing the frame with `<SPC> q f`, the buffer with `README.md`
is still open in the layout's window, as expected.

TLDR: It seems to me that while the layout, open buffer lists and so
on are stored, the info which buffer is open in which window isn't.

#### Reproduction guide :beetle:
- Start Emacs in daemon mode
- Launch a new frame with `emacsclient -c`
- Create a new layout via `<SPC> l 2`
- Open a file in that layout `<SPC> f f`
- Close the frame with `<SPC> q f`
- Open another frame with `emacsclient -c`
- Switch to the previously created layout `<SPC> l 2`
- Observe that the previously loaded buffer is not
  in the window anymore
- Press `<SPC> b b` to observe the buffer is still opened

*Observed behaviour:* :eyes: :broken_heart:
The information about which buffer is loaded in which window is not
stored between frames.

*Expected behaviour:* :heart: :smile:
When I close the last frame/emacsclient and open a new one,
the previously loaded buffers are still shown in the windows they
were in when I closed the previous buffer.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.91
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 7a9a16ec9)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm lsp
      (auto-completion :variables auto-completion-enable-help-tooltip t)
      (c-c++ :variables c-c++-default-mode-for-headers 'c++-mode c-c++-backend 'lsp-ccls c-c++-lsp-enable-semantic-highlight t c-c++-adopt-subprojects t)
      cmake elixir
      (haskell :variables haskell-completion-backend 'lsp)
      emacs-lisp
      (git :variables git-commit-summary-max-length 50)
      (latex :variables latex-build-command "LatexMk")
      (markdown :variables markdown-command "~/.local/bin/pandoc")
      (plantuml :variables plantuml-jar-path "/home/michael/.local/opt/plantuml.jar")
      (python :variables python-backend 'anaconda)
      (semantic)
      (shell-scripts)
      (spacemacs-layouts :variables persp-autokill-buffer-on-remove 'kill-weak)
      (syntax-checking)
      (treemacs)
      (org :variables org-plantuml-jar-path "/home/michael/.local/opt/plantuml.jar")
      yaml)
```
- System configuration features: XAW3D XPM JPEG GIF PNG RSVG CAIRO IMAGEMAGICK DBUS GCONF GLIB NOTIFY INOTIFY ACL GNUTLS FREETYPE ZLIB LUCID X11 XDBE XIM THREADS PDUMPER GMP

#### Possibly relevant info
I have `dotspacemacs-auto-resume-layouts 't` set in my `.spacemacs` config.

**Comment by duianto** [comment]
Confirmed.

But a layouts buffers seem to be saved if one switches layout before killing the frame.

<details><summary>
Windows 10, System Info (Click to expand)
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 7a7b04abb)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey command-log emacs-lisp git haskell helm
            (javascript :variables javascript-backend 'lsp)
            lsp multiple-cursors org
            (python :variables python-backend 'lsp python-lsp-server 'mspyls)
            shell version-control treemacs)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP

</pre>
</details>
